### PR TITLE
Add GlobalMaxPool1D layer

### LINKED ISSUE #62: [Audio] Add GlobalMaxPooling1D layer
We are missing some layers to support the export of models from Keras fully. One of them is the GlobalMaxPooling1D layer.

Add a layer class, write documentation for it, write a test for it, try, if possible, create a small trainable network with it (in your own GitHub) and attach a link here in the comments.

The layer should be placed [here](https://github.com/JetBrains/KotlinDL/tree/master/api/src/main/kotlin/org/jetbrains/kotlinx/dl/api/core/layer/pooling)

As a reference implementation, the [GlovalAvgPool2D layer](https://github.com/JetBrains/KotlinDL/blob/master/api/src/main/kotlin/org/jetbrains/kotlinx/dl/api/core/layer/pooling/GlovalAvgPool2D.kt) could be used, but feel free to improve it!

If some refactoring to pooling layers could be applied, please, try to do it!

Also, support for export and import of layer in JSON format should be added (see [ModelLoader.kt ](https://github.com/JetBrains/KotlinDL/blob/master/api/src/main/kotlin/org/jetbrains/kotlinx/dl/api/inference/keras/ModelLoader.kt) and [ModelSaver.kt](https://github.com/JetBrains/KotlinDL/blob/master/api/src/main/kotlin/org/jetbrains/kotlinx/dl/api/inference/keras/ModelSaver.kt))

A detailed description of the layer can be found [here](https://keras.io/api/layers/pooling_layers/global_max_pooling1d/)

### PR DESCRIPTION
**Pull request by mkaze**
Resolves #62. This PR adds support for `GlobalMaxPool1D` layer.

- [x] Implement layer class
- [x] Add unit tests for layer implementation
- [x] Add support for importing models containing this layer from JSON config files
- [x] Add support for exporting models containing this layer to JSON config files

### CODE REVIEW

File: api/src/main/kotlin/org/jetbrains/kotlinx/dl/api/core/layer/pooling/GlobalMaxPool1D.kt [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+
+/**
+ * Global max pooling operation for 1D temporal data (e.g. audio, timseries, etc.).
+ *
+ * Downsamples the input by taking the maximum value over time dimension.
+ *
+ * @property [dataFormat] The order of dimensions in the input.
+ */
+public class GlobalMaxPool1D(
+    public val dataFormat: String = CHANNELS_LAST,
```
Comment (zaleslaw) [inline api/src/main/kotlin/org/jetbrains/kotlinx/dl/api/core/layer/pooling/GlobalMaxPool1D.kt:24]: Could you please remove the dataFormat

File: api/src/main/kotlin/org/jetbrains/kotlinx/dl/api/core/layer/pooling/GlobalMaxPool1D.kt [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+    name: String = ""
+) : Layer(name) {
+
+    override val hasActivation: Boolean
+        get() = false
+    override val paramCount: Int
+        get() = 0
+    override val weights: Map<String, Array<*>>
+        get() = emptyMap()
```
Comment (zaleslaw) [inline api/src/main/kotlin/org/jetbrains/kotlinx/dl/api/core/layer/pooling/GlobalMaxPool1D.kt:33]: merge with master and add set method

File: api/src/main/kotlin/org/jetbrains/kotlinx/dl/api/core/layer/pooling/GlobalMaxPool1D.kt [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+
+    init {
+        require(dataFormat == CHANNELS_LAST || dataFormat == CHANNELS_FIRST) {
```
Comment (zaleslaw) [inline api/src/main/kotlin/org/jetbrains/kotlinx/dl/api/core/layer/pooling/GlobalMaxPool1D.kt:36]: remove dataFormat

File: api/src/main/kotlin/org/jetbrains/kotlinx/dl/api/core/layer/pooling/GlobalMaxPool1D.kt [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+            "The dataFormat should be either \"$CHANNELS_LAST\" or \"$CHANNELS_FIRST\"."
+        }
+    }
+
+    override fun build(tf: Ops, kGraph: KGraph, inputShape: Shape) {}
+
+    override fun computeOutputShape(inputShape: Shape): Shape {
+        return if (dataFormat == CHANNELS_LAST) {
```
Comment (zaleslaw) [inline api/src/main/kotlin/org/jetbrains/kotlinx/dl/api/core/layer/pooling/GlobalMaxPool1D.kt:44]: remove data format

File: api/src/main/kotlin/org/jetbrains/kotlinx/dl/api/core/layer/pooling/GlobalMaxPool1D.kt [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+        }
+    }
+
+    override fun forward(
+        tf: Ops,
+        input: Operand<Float>,
+        isTraining: Operand<Boolean>,
+        numberOfLosses: Operand<Float>?
+    ): Operand<Float> {
+        return if (dataFormat == CHANNELS_LAST) {
```
Comment (zaleslaw) [inline api/src/main/kotlin/org/jetbrains/kotlinx/dl/api/core/layer/pooling/GlobalMaxPool1D.kt:57]: remove dataFormat

File: api/src/main/kotlin/org/jetbrains/kotlinx/dl/api/core/layer/pooling/GlobalMaxPool1D.kt [hunk 6]
```diff
@@ -0,0 +1,66 @@
...
+            tf.max(input, tf.constant(1))
+        } else {
+            tf.max(input, tf.constant(2))
+        }
+    }
+
+    override fun toString(): String =
+        "GlobalMaxPool1D(dataFormat=$dataFormat)"
```
Comment (zaleslaw) [inline api/src/main/kotlin/org/jetbrains/kotlinx/dl/api/core/layer/pooling/GlobalMaxPool1D.kt:65]: Maybe add name of the layer

File: api/src/test/kotlin/org/jetbrains/kotlinx/dl/api/core/layer/GlobalMaxPool1DTest.kt [hunk 7]
```diff
@@ -0,0 +1,113 @@
...
+            arrayOf(
+                floatArrayOf(5.0f, 3.0f, 1.0f),
+                floatArrayOf(6.0f, -2.5f, 4.0f),
+            )
+        )
+        val expected = arrayOf(
+            floatArrayOf(3.0f, 5.0f),
+            floatArrayOf(5.0f, 6.0f)
+        )
+        val layer = GlobalMaxPool1D(dataFormat = CHANNELS_FIRST)
```
Comment (zaleslaw) [inline api/src/test/kotlin/org/jetbrains/kotlinx/dl/api/core/layer/GlobalMaxPool1DTest.kt:83]: remove this test due to removed dataFormat

**Review by zaleslaw — CHANGES_REQUESTED** [review]
Hi, thanks for the PR; please, remove dataFormat field as we discussed and merge it with master

File: api/src/main/kotlin/org/jetbrains/kotlinx/dl/api/core/layer/pooling/GlobalMaxPool1D.kt (hunk 1 above)
Comment (mkaze) [reply api/src/main/kotlin/org/jetbrains/kotlinx/dl/api/core/layer/pooling/GlobalMaxPool1D.kt]: Done!

File: api/src/main/kotlin/org/jetbrains/kotlinx/dl/api/core/layer/pooling/GlobalMaxPool1D.kt (hunk 2 above)
Comment (mkaze) [reply api/src/main/kotlin/org/jetbrains/kotlinx/dl/api/core/layer/pooling/GlobalMaxPool1D.kt:28]: Done!

File: api/src/main/kotlin/org/jetbrains/kotlinx/dl/api/core/layer/pooling/GlobalMaxPool1D.kt (hunk 3 above)
Comment (mkaze) [reply api/src/main/kotlin/org/jetbrains/kotlinx/dl/api/core/layer/pooling/GlobalMaxPool1D.kt]: Done!

File: api/src/main/kotlin/org/jetbrains/kotlinx/dl/api/core/layer/pooling/GlobalMaxPool1D.kt (hunk 4 above)
Comment (mkaze) [reply api/src/main/kotlin/org/jetbrains/kotlinx/dl/api/core/layer/pooling/GlobalMaxPool1D.kt]: Done!

File: api/src/main/kotlin/org/jetbrains/kotlinx/dl/api/core/layer/pooling/GlobalMaxPool1D.kt (hunk 5 above)
Comment (mkaze) [reply api/src/main/kotlin/org/jetbrains/kotlinx/dl/api/core/layer/pooling/GlobalMaxPool1D.kt]: Done!

File: api/src/main/kotlin/org/jetbrains/kotlinx/dl/api/core/layer/pooling/GlobalMaxPool1D.kt (hunk 6 above)
Comment (mkaze) [reply api/src/main/kotlin/org/jetbrains/kotlinx/dl/api/core/layer/pooling/GlobalMaxPool1D.kt]: Done!

File: api/src/test/kotlin/org/jetbrains/kotlinx/dl/api/core/layer/GlobalMaxPool1DTest.kt (hunk 7 above)
Comment (mkaze) [reply api/src/test/kotlin/org/jetbrains/kotlinx/dl/api/core/layer/GlobalMaxPool1DTest.kt]: Done!

**Pull request merged**